### PR TITLE
DOXYGEN: doxygen review on remaining common headers

### PR DESCRIPTION
**Pull request by b-gent**
The PR covers the remaining high-priority headers from the common folder.

- textconsole.h
- timer.h
- translation.h
- ustr.h
- util.h

HTML output available here:
https://b-gent.github.io/2627/d7/da5/group__common__text__console.html

### CODE REVIEW

**Comment by b-gent** [comment]
I'll fix the conflict tomorrow.

**Comment by b-gent** [comment]
This is ready for review @sev-  @criezy

**Review by criezy — CHANGES_REQUESTED** [review]
First pass on `textconsole.h`, `timer.h`, and `translation.h`.
I will look at the other two files later this week.

File: common/translation.h [hunk 1]
```diff
@@ -50,22 +50,31 @@ enum TranslationIDs {
...
+ * Structure describing a translation language.
+ */
 struct TLanguage {
-	U32String name;
-	int id;
+	U32String name; /*!< Language name string. */
+	int id;         /*!< Language ID. */
 
 	TLanguage() : id(0) {}
+	/**
```
Comment (mgerhardy) [inline common/translation.h:61]: This looks like a whitespace issue

File: common/translation.h [hunk 2]
```diff
@@ -50,22 +50,31 @@ enum TranslationIDs {
...
 
+/**
+ * Structure describing a message to be translated.
+ */
 struct PoMessageEntry {
-	int msgid;
-	String msgctxt;
-	U32String msgstr;
+	int msgid;         /*!< ID of the message. */
+	String msgctxt;    /*!< @todo Doc required. */
```
[the remark this answers is not in the document]
Comment (mgerhardy) [reply common/translation.h:76]: Can be used to solving ambiguities. Here are more details about it: https://www.gnu.org/software/gettext/manual/html_node/Contexts.html

File: common/ustr.h [hunk 3]
```diff
@@ -77,76 +77,107 @@ class U32String : public BaseString<u32char_type_t> {
...
-	void insertString(const char *s, uint32 p, CodePage page = kUtf8);
-	void insertString(const String &s, uint32 p, CodePage page = kUtf8);
+	void insertString(const char *s, uint32 p, CodePage page = kUtf8);   /*!< Insert string @p s into this string at position @p p. */
+	void insertString(const String &s, uint32 p, CodePage page = kUtf8); /*!< @overload */
 
 	/** Return a substring of this string */
 	U32String substr(size_t pos = 0, size_t len = npos) const;
 
-	const uint32 *u32_str() const {
+	const uint32 *u32_str() const {   /*!< @todo Doc required (Return the number of characters in the string???) */
```
Comment (mgerhardy) [inline common/ustr.h:161]: Returns the string as UTF-32 pointer

File: common/util.h [hunk 4]
```diff
@@ -226,45 +252,51 @@ bool isUpper(int c);
...
  * largest suitable units. For example, 1474560 bytes as 1.4 MB.
  * 
- * @param bytes		size in bytes to be represented
- * @param unitsOut	(out-parameter) string with units
- * @note			use _() to translate units correctly
- * @return			string with a floating point number representing given size
+ * @param[in]  bytes     Size in bytes to be represented.
+ * @param[out] unitsOut  String with units.
+ *
+ * @note Use _() to translate units correctly.
```
Comment (mgerhardy) [inline common/util.h:297]: should this maybe be prefixed by `@c` (`@c _()`)

**Comment by b-gent** [comment]
Comments fixed, thank you for reviews @criezy  and @mgerhardy 
This should be good to go now.

**Review by criezy — APPROVED** [review]
Thank you. This looks good now.
There is one minor issue that I will fix when merging.

**Comment by criezy** [comment]
And merged.

**Pull request closed without merging**